src/client/reducers/user/types.ts

```diff
@@ -0,0 +1,32 @@
+export type UrlType = {
+  shortUrl: string
+  longUrl: string
+}
+
+export type UserState = {
+  initialised: boolean
+  isFetchingUrls: boolean
+  urls: Array<UrlType>
+  shortUrl: string
+  longUrl: string
+  createUrlModal: boolean
+  tableConfig: UrlTableConfig
+  urlCount: 0
+}
+
+export type UrlTableConfig = {
+  numberOfRows: number
+  pageNumber: number
+  sortDirection: SortDirection
+  orderBy: string
+  searchText: string
+  filter: {
+    isFile?: boolean
+    orderBy?: string
+    state?: UrlState
+  }
+}
+
+export type UrlState = 'ACTIVE' | 'INACTIVE'
+
+export type SortDirection = 'desc' | 'asc'
```

Comment:
both `UrlState` and `SortDirection` (and also all other types following this pattern) are better represented in a typescript enum